src/develop/develop.c

```diff
@@ -743,6 +746,11 @@ void dt_dev_pop_history_items(dt_develop_t *dev, int32_t cnt)
   // printf("dev popping all history items >= %d\n", cnt);
   dt_pthread_mutex_lock(&dev->history_mutex);
   darktable.gui->reset = 1;
+  if(cnt != dev->history_end)
+  {
+      //History really changes -> invalidate mipmaps
+      dev->history_was_changed = TRUE;
```

Comment:
Will not work when count of history items does not change.
https://github.com/darktable-org/darktable/pull/959/files#diff-79a3db631cb5ea0369a221a5c31a9eddL598